### DISCUSSION TITLE
Rescale the AR Drone object in UNreal Engine

### DISCUSSION THREAD

**Discussion by Moj-Dev** [microsoft/AirSim#3262] [Support Q&A] [answered]
How can I rescale the quadrotor(player) object's size in Unreal Engine?

**Comment by ahmed-elsaharti** [comment, marked as answer]
Hi @Moj-Dev, the simplest way I found to rescale AirSim's quadrotor in the **Blocks** environment (I'm pretty sure doing this with a custom environment would be similar too) would be to edit the BP_FlyingPawn asset by:
1. Going to the contents folder in the UE editor's content browser:
![image](https://user-images.githubusercontent.com/49626509/103326155-a4d29e80-4a14-11eb-87b6-1257c01221c9.png)
2. Doubleclicking on AirSimAssets:
![image](https://user-images.githubusercontent.com/49626509/103326185-c16ed680-4a14-11eb-9775-9cb148dd040e.png)
This will open the AirSimAssets 'level':
![image](https://user-images.githubusercontent.com/49626509/103326213-e2cfc280-4a14-11eb-8199-775562add008.png)
3. Now click on click on `Edit BP_FlyingPawn` under the World Outliner:
![image](https://user-images.githubusercontent.com/49626509/103326251-07c43580-4a15-11eb-9a9e-4991c4faa03b.png)
4. This will open the default BP_FlyingPawn asset (which is actually located under Plugins\AirSim\Content\Blueprints):
![image](https://user-images.githubusercontent.com/49626509/103326283-3a6e2e00-4a15-11eb-97d3-09b8255bb0ad.png)
5. From there, click on viewport:
![image](https://user-images.githubusercontent.com/49626509/103326296-5671cf80-4a15-11eb-84f1-55ec90ade3f5.png)
![image](https://user-images.githubusercontent.com/49626509/103326323-7d300600-4a15-11eb-85da-9397784d1cef.png)
6. Then you click on the drone's body (ie: select the bodymesh):
![image](https://user-images.githubusercontent.com/49626509/103326348-95a02080-4a15-11eb-9f73-2009af85233f.png)
![image](https://user-images.githubusercontent.com/49626509/103326358-9e90f200-4a15-11eb-84df-b515a6feff8f.png)
7. You can then scale it up or down in X and/or Y and/or Z:
![image](https://user-images.githubusercontent.com/49626509/103326401-c6805580-4a15-11eb-8b6a-ece3102e9297.png)
![image](https://user-images.githubusercontent.com/49626509/103326415-d9932580-4a15-11eb-946c-efb0a72653a7.png)
(the props and cameras will scale up/down accordingly as far as i remember)
Then you save, compile and run your simulation (to go back to the main AirSim level in Blocks that you need to run (ie: not AirSimAssets) go to the FlyingCPP folder in the content browser then the Maps folder and double-click the FlyingExampleMap level):
![image](https://user-images.githubusercontent.com/49626509/103326630-e7957600-4a16-11eb-9d02-a4a6f9c54eb2.png)

Hope this helps ^_^

**Reply by Moj-Dev** [reply]
> Hi @Moj-Dev, the simplest way I found to rescale AirSim's quadrotor in the **Blocks** environment (I'm pretty sure doing this with a custom environment would be similar too) would be to edit the BP_FlyingPawn asset by:
> 
> 1. Going to the contents folder in the UE editor's content browser:
>    ![image](https://user-images.githubusercontent.com/49626509/103326155-a4d29e80-4a14-11eb-87b6-1257c01221c9.png)
> 2. Doubleclicking on AirSimAssets:
>    ![image](https://user-images.githubusercontent.com/49626509/103326185-c16ed680-4a14-11eb-9775-9cb148dd040e.png)
>    This will open the AirSimAssets 'level':
>    ![image](https://user-images.githubusercontent.com/49626509/103326213-e2cfc280-4a14-11eb-8199-775562add008.png)
> 3. Now click on click on `Edit BP_FlyingPawn` under the World Outliner:
>    ![image](https://user-images.githubusercontent.com/49626509/103326251-07c43580-4a15-11eb-9a9e-4991c4faa03b.png)
> 4. This will open the default BP_FlyingPawn asset (which is actually located under Plugins\AirSim\Content\Blueprints):
>    ![image](https://user-images.githubusercontent.com/49626509/103326283-3a6e2e00-4a15-11eb-97d3-09b8255bb0ad.png)
> 5. From there, click on viewport:
>    ![image](https://user-images.githubusercontent.com/49626509/103326296-5671cf80-4a15-11eb-84f1-55ec90ade3f5.png)
>    ![image](https://user-images.githubusercontent.com/49626509/103326323-7d300600-4a15-11eb-85da-9397784d1cef.png)
> 6. Then you click on the drone's body (ie: select the bodymesh):
>    ![image](https://user-images.githubusercontent.com/49626509/103326348-95a02080-4a15-11eb-9f73-2009af85233f.png)
>    ![image](https://user-images.githubusercontent.com/49626509/103326358-9e90f200-4a15-11eb-84df-b515a6feff8f.png)
> 7. You can then scale it up or down in X and/or Y and/or Z:
>    ![image](https://user-images.githubusercontent.com/49626509/103326401-c6805580-4a15-11eb-8b6a-ece3102e9297.png)
>    ![image](https://user-images.githubusercontent.com/49626509/103326415-d9932580-4a15-11eb-946c-efb0a72653a7.png)
>    (the props and cameras will scale up/down accordingly as far as i remember)
>    Then you save, compile and run your simulation (to go back to the main AirSim level in Blocks that you need to run (ie: not AirSimAssets) go to the FlyingCPP folder in the content browser then the Maps folder and double-click the FlyingExampleMap level):
>    ![image](https://user-images.githubusercontent.com/49626509/103326630-e7957600-4a16-11eb-9d02-a4a6f9c54eb2.png)
> 
> Hope this helps ^_^

Thank you so much!
